content/en/docs/setup/best-practices/certificates.md

```diff
@@ -77,7 +77,7 @@ Required certificates:
 
 | Default CN                    | Parent CA                 | O (in Subject) | kind                                   | hosts (SAN)                                 |
 |-------------------------------|---------------------------|----------------|----------------------------------------|---------------------------------------------|
-| kube-etcd                     | etcd-ca                   |                | server, client                         | `localhost`, `127.0.0.1`                        |
+| kube-etcd                     | etcd-ca                   |                | server, client                         | `<hostname>`, `<Host_IP>`, `localhost`, `127.0.0.1` |
```

Comment:
this seems topology specific but if kube-etcd is the serving cert for etcd, kubeadm at least has the hostname and host IP in the SAN.

/lgtm
/sig auth cluster-lifecycle

Reply:
I am not sure I understand? The etcd instance running in the cluster is checking the SAN provided by the client(kubeadm). 

 In the bare metal cluster setup with a self signed ca certificate etcd refuses the certificate since kubeadm is using the kube-etcd as a client certificate to connect to the etcd instance with its `<Host_IP>` during the **phase etcd-check**.  

The error message of etcd on the cluster was something along the lines of `connection refused <Host IP of new control plane trying to join the cluster> tls: bad certificate`. If you need I can recreate my cluster and copy the error messages by both parties.

The `<hostname>` would be required as a SAN if an etcd member endpoint would be defined as a hostname instead of an IP. 

In my given case when etcd is running as pods from the manifests in /etc/kubernetes/manifests the `<Host_IP>` was required for me to be set in the SAN of the kube-etcd certificate which is used by kubeadm when joining a new control plane.

Reply:
Note also that last time I checked the generated certificates in kubernetes that it is adding `<Host_IP>` and `<hostname>` as a SAN in the kube-etcd certificate when kubernetes is managing its pki environment.

Reply:
> In my given case when etcd is running as pods from the manifests in /etc/kubernetes/manifests the <Host_IP> was required for me to be set in the SAN of the kube-etcd certificate which is used by kubeadm when joining a new control plane.

like i said, kubeadm adds the hostname and host IP in the serving certficate by default. but that happens in the stacked / local etcd case, where etcd is managed by kubeadm and runs as a static pod from /etc/kubernetes/manifests

but if you are running etcd as external (from kubeadm's perspective) or behind a etcd load balancer, then adding a particular host in the SANs might not be what you want, therefore this is topology specific.

> In the bare metal cluster setup with a self signed ca certificate etcd refuses the certificate since kubeadm is using the kube-etcd as a client certificate to connect to the etcd instance with its <Host_IP> during the phase etcd-check.

kubeadm signs a specific certificate for health checks from clients `healthcheck-client.crt`:
(it is also listed in the table here.)

```
controlplane $ ls /etc/kubernetes/pki/etcd/
ca.crt  ca.key  healthcheck-client.crt  healthcheck-client.key  peer.crt  peer.key  server.crt  server.key
```

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/phases/join/checketcd.go#L69
https://github.com/kubernetes/kubernetes/blob/0153febd9f0098d4b8d0d484927710eaf899ef40/cmd/kubeadm/app/phases/etcd/local.go#L75
https://github.com/kubernetes/kubernetes/blob/0153febd9f0098d4b8d0d484927710eaf899ef40/cmd/kubeadm/app/util/etcd/etcd.go#L107-L112

Reply:
Thank you for clarifying. 

> but if you are running etcd as external (from kubeadm's perspective) or behind a etcd load balancer, then adding a particular host in the SANs might not be what you want, therefore this is topology specific.

My setup was external ca, stacked etcd(generated by kubeadm). I was under the impression that the documentation was guiding along under this assumption as well since the table with the requirements to get a working cluster is for external ca.

If that is not the case then this PR can be gladly closed since there is already a [note](https://github.com/kubernetes/website/pull/30716/files#diff-e237abb140212a8c4c0198eeaef9fa26b3a35b4d71e8588174cf7ff825b2d41aL100) about adding additional SAN's for specific setups.

Reply:
since i already LGTM-ed - i think the addition is fine.